shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java

```diff
@@ -87,41 +145,52 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
    * @param filePath The {@link AlluxioURI} path to load into Alluxio
    * @param local whether to load data to local worker even when the data is already loaded remotely
    */
-  private void load(AlluxioURI filePath, boolean local) throws AlluxioException, IOException {
+  private void load(AlluxioURI filePath, boolean local, List<LoadTask> tasks)
+      throws AlluxioException, IOException {
     URIStatus status = mFileSystem.getStatus(filePath);
+
     if (status.isFolder()) {
       List<URIStatus> statuses = mFileSystem.listStatus(filePath);
       for (URIStatus uriStatus : statuses) {
         AlluxioURI newPath = new AlluxioURI(uriStatus.getPath());
-        load(newPath, local);
+        load(newPath, local, tasks);
       }
     } else {
-      OpenFilePOptions options =
-          OpenFilePOptions.newBuilder().setReadType(ReadPType.CACHE).build();
       if (local) {
         if (!mFsContext.hasNodeLocalWorker()) {
-          System.out.println("When local option is specified,"
-              + " there must be a local worker available");
+          System.out.println(
+              "When local option is specified," + " there must be a local worker available");
           return;
         }
       } else if (status.getInAlluxioPercentage() == 100) {
         // The file has already been fully loaded into Alluxio.
         System.out.println(filePath + " already in Alluxio fully");
         return;
       }
-      Closer closer = Closer.create();
-      try {
-        FileInStream in = closer.register(mFileSystem.openFile(filePath, options));
-        byte[] buf = new byte[8 * Constants.MB];
-        while (in.read(buf) != -1) {
-        }
-      } catch (Exception e) {
-        throw closer.rethrow(e);
-      } finally {
-        closer.close();
+
+      queueLoadTasks(filePath, status, local, tasks);
+    }
+  }
+
+  private void queueLoadTasks(AlluxioURI filePath, URIStatus status, boolean local,
```

Comment:
I think this method is naive in that, if somebody wants to load a very large file-tree, `tasks` list here will grow quite large.